src/main/java/jenkins/plugins/openstack/compute/ServerScope.java

```diff
@@ -148,33 +149,57 @@ protected boolean _equals(ServerScope o) {
      * Server marked as <tt>node:asdf</tt> will live only to support Jenkins node <tt>asdf</tt>.
      */
     public static final class Node extends ServerScope {
+        private final @Nonnull String name;
+        private final @CheckForNull Integer cloudStatsFingerprint;
 
-        public Node(String specifier) {
+        public Node(@Nonnull String specifier) {
             super("node", specifier);
+
+            String[] split = specifier.split(":");
+            switch (split.length) {
+                case 1:
+                    name = specifier;
+                    cloudStatsFingerprint = null;
+                break;
+                case 2:
+                    name = split[0];
+                    cloudStatsFingerprint = Integer.parseInt(split[1]);
+                break;
+                default: throw new IllegalArgumentException("Invalid node scope specifier " + specifier);
+            }
+        }
+
+        public Node(@Nonnull String serverName, @Nonnull ProvisioningActivity.Id id) {
+            super("node", serverName+":"+id.getFingerprint());
+            name = serverName;
+            cloudStatsFingerprint = id.getFingerprint();
         }
 
-        public String getName() {
-            return specifier;
+        @Nonnull public String getName() {
+            return name;
         }
 
         @Override
         public boolean isOutOfScope(@Nonnull Server server) {
-            if (Jenkins.get().getNode(specifier) != null) return false;
-
-            // The node may be provisioned or deleted at the moment - do not interfere
-            for (ProvisioningActivity pa : CloudStatistics.get().getActivities()) {
-                if (specifier.equals(pa.getName())) {
```

Comment:
This check had no effect as the activity name was not set to {{nodeName}}/{{specifier}} until provisioning was completed. Making the actual, unique cloud-stats fingerprint part of the scope specifier so we can reliably tie OS Servers to ProvisioningActivities.